Sources/MapboxSearch/PublicAPI/Use Cases/Place Autocomplete/PlaceAutocomplete.swift

```diff
@@ -291,52 +291,28 @@ private extension PlaceAutocomplete {
         let filteredSuggestions = suggestions.filter {
             !$0.resultTypes.contains(.category) && !$0.resultTypes.contains(.query)
         }
-        guard !filteredSuggestions.isEmpty else {
-            return completion(.success([]))
-        }
-        
-        let dispatchGroup = DispatchGroup()
-        var resolvingError: Error?
-        
-        var resolvedSuggestions: [Suggestion?] = Array(repeating: nil, count: filteredSuggestions.count)
-        let lock = NSLock()
-
-        filteredSuggestions.enumerated().forEach { iterator in
-            dispatchGroup.enter()
 
-            if iterator.element.center != nil {
-                do {
-                    let resolvedSuggestion = try Suggestion.from(searchSuggestion: iterator.element, options: options)
-                    lock.sync {
-                        resolvedSuggestions[iterator.offset] = resolvedSuggestion
-                    }
-                } catch {
-                    resolvingError = error
-                }
-                dispatchGroup.leave()
-            } else {
-                retrieve(suggestion: iterator.element, with: options) { result in
-                    defer { dispatchGroup.leave() }
+        let resolvedSuggestions = filteredSuggestions.compactMap { result -> Suggestion? in
+            let matchingName = result.matchingName ?? ""
+            let distance = result.distance.flatMap { CLLocationDistance(integerLiteral: $0.int64Value) }
+            let coreResultTypes = result.types.compactMap { CoreResultType(rawValue: $0.intValue) }
+            let placeTypes = SearchResultType(coreResultTypes: coreResultTypes)
+            let categories = result.categories ?? []
+            let routablePoints = result.routablePoints?.compactMap { RoutablePoint(routablePoint: $0) } ?? []
+            let underlying: Suggestion.Underlying = .suggestion(result, options)
 
-                    switch result {
-                    case .success(let suggestion):
-                        lock.sync {
-                            resolvedSuggestions[iterator.offset] = suggestion
-                        }
-                    case .failure(let error):
-                        resolvingError = error
-                    }
-                }
-            }
-        }
-        
-        dispatchGroup.notify(queue: .main) {
-            let results = resolvedSuggestions.compactMap({ $0 })
-            if results.isEmpty {
-                completion(.failure(resolvingError ?? SearchError.responseProcessingFailed))
-            } else {
-                completion(.success(results))
-            }
+            return Suggestion(name: matchingName,
```

Comment:
From what I understand,`name` and `matchingName` are different properties.
```
/** Names matched to the \a languages array. */
@property (nonatomic, readonly, nonnull, copy) NSArray<NSString *> *names;
/** The feature name, as matched by the search algorithm. */
@property (nonatomic, readonly, nullable, copy) NSString *matchingName;
```

Before this pull request the values `searchSuggestion.names.first ?? ""` was used, now `matchingName` is used. Is this change expected?

Reply:
Oh that's probably better to use `name`